kubeconfig_plugin.py

```diff
@@ -111,63 +109,107 @@ class ErrorOutput:
     id="kubeconfig",
     name="kubeconfig plugin",
     description=(
-        "Inputs a kubeconfig, parses it and extracts the kubernetes cluster"
-        " details "
+            "Inputs a kubeconfig, parses it and extracts the kubernetes cluster"
+            " details "
     ),
     outputs={"success": SuccessOutput, "error": ErrorOutput},
 )
 def extract_kubeconfig(
-    params: InputParams,
+        params: InputParams,
 ) -> typing.Tuple[str, typing.Union[SuccessOutput, ErrorOutput]]:
-
     print("==>> Parsing and extracting kubernetes cluster details ...")
 
     try:
         kubeconfig = yaml.safe_load(params.kubeconfig)
-        kcl = config.kube_config.KubeConfigLoader(kubeconfig)
 
-        output = {}
-        output["cluster_name"] = kcl._current_context["context"]["cluster"]
-        output["server_url"] = kcl._cluster.value["server"]
-        output["certificate_authority_data"] = kcl._cluster.value[
-            "certificate-authority-data"
-        ]
-        output["user"] = kcl._current_context["context"]["user"]
+        try:
+            if kubeconfig["kind"] != "Config":
+                return "error", ErrorOutput("The provided file is not a kubeconfig file")
+        except KeyError:
+            return "error", ErrorOutput("The provided file is not a kubeconfig file (missing 'kind' field)")
 
         try:
-            output["client_certificate_data"] = kcl._user.value[
-                "client-certificate-data"
-            ]
-            output["client_key_data"] = kcl._user.value["client-key-data"]
-        except Exception:
-            print(
-                "client certificate and key data missing, trying to extract"
-                " token"
-            )
+            current_context = kubeconfig["current-context"]
+        except KeyError:
+            return "error", ErrorOutput(
+                "The provided kubeconfig file does not have a current-context set. Please set a current context to use.")
```

Comment:
Consider the following to reduce the error returns and lines of code a bit:

```
kind = kubeconfig.get("kind", "")
if kind != "Config":
    addendum = "" if kind else " (missing 'kind' field)"
    return "error", ErrorOutput(f"The provided file is not a kubeconfig file{addendum}")
current_context = kubeconfig.get("current-context", "")
if not current_context:
    return "error", ErrorOutput(
                "The provided kubeconfig file does not have a current-context set. Please set a current context to use.")
```

Reply:
Code refactored.